subprojects/launcher/src/test/groovy/org/gradle/launcher/cli/converter/PropertiesToStartParameterConverterTest.groovy

```diff
@@ -40,4 +41,27 @@ class PropertiesToStartParameterConverterTest extends Specification {
         then:
         thrown(IllegalArgumentException)
     }
+
+    def "converts log levels"() {
+        expect:
+        converter.convert([(LOG_LEVEL_PROPERTY): "DEBUG"], new StartParameter()).logLevel == LogLevel.DEBUG
+        converter.convert([(LOG_LEVEL_PROPERTY): "Info"], new StartParameter()).logLevel == LogLevel.INFO
+        converter.convert([(LOG_LEVEL_PROPERTY): "LifeCycle"], new StartParameter()).logLevel == LogLevel.LIFECYCLE
+        converter.convert([(LOG_LEVEL_PROPERTY): "warn"], new StartParameter()).logLevel == LogLevel.WARN
+        converter.convert([(LOG_LEVEL_PROPERTY): "quiet"], new StartParameter()).logLevel == LogLevel.QUIET
```

Comment:
It's perfectly fine how you did it. You can also look into the [`where` block documentation](http://spockframework.org/spock/docs/1.1-rc-4/spock_primer.html).